### DISCUSSION TITLE
Warning with decorators.

### DISCUSSION THREAD

**Discussion by AntoineD** [mkdocstrings/griffe#243] [Q&A] [answered]
When building the docs from this [repo](https://gitlab.com/MatthiasDeLozzo/gemseo-mlearning/-/tree/16), warnings are issued on a decorated [method](https://gitlab.com/MatthiasDeLozzo/gemseo-mlearning/-/blob/16/src/gemseo_mlearning/adaptive/distribution.py#L238):
```
WARNING -  griffe: src/gemseo_mlearning/adaptive/distribution.py:258: No type or annotation for parameter 'algo'
WARNING -  griffe: src/gemseo_mlearning/adaptive/distribution.py:258: Parameter 'algo' does not appear in the function
           signature
WARNING -  griffe: src/gemseo_mlearning/adaptive/distribution.py:260: No type or annotation for parameter '*args'
WARNING -  griffe: src/gemseo_mlearning/adaptive/distribution.py:260: Parameter '*args' does not appear in the function
           signature
WARNING -  griffe: src/gemseo_mlearning/adaptive/distribution.py:261: No type or annotation for parameter '**kwargs'
WARNING -  griffe: src/gemseo_mlearning/adaptive/distribution.py:261: Parameter '**kwargs' does not appear in the
           function signature
```
The related [decorator](https://gitlab.com/gemseo/dev/gemseo/-/blob/develop/src/gemseo/mlearning/data_formatters/supervised_data_formatters.py?ref_type=heads#L58) has type annotations. 

Could this be a bug in griffe?

Repro with `tox -e doc`.

**Comment by pawamoy** [comment, marked as answer]
Hello, I'd say it is caused by a combination of your wrapper and your docstring-inheritance extension. I'd recommend using `functools.wraps` on your `wrapper` function, see https://mkdocstrings.github.io/troubleshooting/#my-wrapped-function-shows-documentationcode-for-its-wrapper-instead-of-its-own. Otherwise your function docstrings is overwritten by the wrapper docstring.

Also, check out our official docstring-inheritance extension: https://mkdocstrings.github.io/griffe-inherited-docstrings/. It's probably less specific than your own though, but maybe will work for you too. One benefit is that it works statically too, so won't require dynamically importing objects like you do in your own extension.

**Reply by AntoineD** [reply]
Thanks, I'll definitely add `functools.wraps` to the wrappers.

I've seen the official docstring-inheritance extension when it was release, unfortunately it does not fit here.

**Reply by AntoineD** [reply]
Using `griffe-inherited-docstrings` could actually be complementary to my wrapper.
When griffe runs extensions, is it sure to be in the same order in which the extensions are declared in `mkdocs.yml`?

**Reply by pawamoy** [reply]
Yes, they run in the same order :slightly_smiling_face:
